be/src/formats/parquet/encoding_dict.h

```diff
@@ -221,6 +221,13 @@ class DictDecoder<Slice> final : public Decoder {
         }
         case VALUE: {
             raw::stl_vector_resize_uninitialized(&_slices, count);
+            size_t size = _dict.size();
+            for (int i = 0; i < count; ++i) {
+                if (_indexes[i] >= size) {
+                    return Status::InternalError(
+                            fmt::format("dict code is out of range. code = {}, size = {}", _indexes[i], size));
+                }
+            }
             for (int i = 0; i < count; ++i) {
```

Comment:
Why not check `_indexes[i]` before this line and avoid introducing an extra for-loop?

Reply:
to avoid branch on decode stage.